File: poc-kronos-frontend/src/model/avaliacao/Avaliacao.js
class Avaliacao {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.horaInicio = dados ? dados.horaInicio : null;
    this.grupo = dados ? dados.grupo : null;
    this.numeroMaxParticipante = dados ? dados.numeroMaxParticipante : null;
    this.peso = dados ? dados.peso : null;
    this.observacao = dados ? dados.observacao : null;
    this.anulada = dados ? dados.anulada : null;
    this.motivoAnulacao = dados ? dados.motivoAnulacao : null;
    this.idTipoFormato = dados ? dados.idTipoFormato : null;
    this.nomeTipoFormato = dados ? dados.nomeTipoFormato : null;
    this.idTipoRegistroNota = dados ? dados.idTipoRegistroNota : null;
    this.idTipoAbrangencia = dados ? dados.idTipoAbrangencia : null;
    this.nomeTipoAbrangencia = dados ? dados.nomeTipoAbrangencia : null;
    this.idAtividade = dados ? dados.idAtividade : null;
    this.idTipoAvaliacao = dados ? dados.idTipoAvaliacao : null;
    this.nomeTipoAvaliacao = dados ? dados.nomeTipoAvaliacao : null;
    this.dataPrevista = dados ? dados.dataPrevista : null;
    this.tempoDuracao = dados ? dados.tempoDuracao : null;
    this.qtdAvaliados = dados ? dados.qtdAvaliados : null;
    this.gruposAvaliacoes = dados ? dados.gruposAvaliacoes : [];
    this.avaliacoesHabilidades = dados ? dados.avaliacoesHabilidades : [];
    this.idTurma = dados ? dados.idTurma : null;
    this.idDisciplina = dados ? dados.idDisciplina : null;
  }

  static validations(required, maxLength, numeric) {
    return {
      idAtividade: { required },
      peso: { required },
      horaInicio: { required },
      numeroMaxParticipante: { required, numeric },
      idTipoFormato: { required },
      idTipoRegistroNota: { required },
      idTipoAbrangencia: { required },
      idTipoAvaliacao: { required },
      tempoDuracao: { required, numeric },
      observacao: { maxLength: maxLength(200) },
      motivoAnulacao: { maxLength: maxLength(200) },
      idTurma: { required },
      idDisciplina: { required }
    };
  }

  static validationMessages() {
    return {
      idAtividade: {
        required: "Atividade ?? obrigat??ria"
      },
      peso: {
        required: "Peso ?? obrigat??rio"
      },
      horaInicio: {
        required: "Hora Inc??o ?? obrigat??ria"
      },
      numeroMaxParticipante: {
        required: "N??mero m??ximo de participantes ?? obrigat??rio",
        numeric: "Somente n??meros"
      },
      idTipoFormato: {
        required: "Formato ?? obrigat??rio"
      },
      idTipoRegistroNota: {
        required: "Registro de Nota ?? obrigat??ria"
      },
      idTipoAbrangencia: {
        required: "Abrang??ncia ?? obrigat??ria"
      },
      idTipoAvaliacao: {
        required: "Avalia????o ?? obrigat??ria"
      },
      tempoDuracao: {
        required: "Tempo de dura????o ?? obrigat??ria",
        numeric: "Somente n??meros"
      },
      observacao: {
        maxLength: "Observa????o s?? aceita no m??ximo 200 caracteres"
      },
      motivoAnulacao: {
        maxLength: "Motivo Anula????o s?? aceita no m??ximo 200 caracteres"
      },
      idTurma: {
        required: "Turma ?? obrigat??ria"
      },
      idDisciplina: {
        required: "Disciplina ?? obrigat??ria"
      }
    };
  }
}
export default Avaliacao;

File: src/main/java/br/com/kronos/backend/aplicacao/avaliacao/Avaliacao.java

package br.com.kronos.backend.aplicacao.avaliacao;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalTime;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Avaliacao {
	
	public static final String COMBO_CACHE_NAME = "combo_avaliacoes";
    
    private long id;
    private LocalTime horaInicio;
    private boolean grupo;
    private Integer numeroMaxParticipante;
    private Integer peso;
    private String observacao;
    private boolean anulada;
    private String motivoAnulacao;
    private Integer idTipoFormato;
    private Integer idTipoRegistroNota;
    private Integer idTipoAbrangencia;
    private Long idAtividade;
    private Integer idTipoAvaliacao;
    private Integer tempoDuracao;
	private Long idTurma;
	private Long idDisciplina;

    @Builder
    public Avaliacao(long id, LocalTime horaInicio, boolean grupo, Integer numeroMaxParticipante, Integer peso, String observacao, 
                     boolean anulada,  String motivoAnulacao, Integer idTipoFormato, Integer idTipoRegistroNota, Integer idTipoAbrangencia, 
                     Long idAtividade, Integer idTipoAvaliacao, Integer tempoDuracao, Long idTurma, Long idDisciplina) {

        this.id = id;
        this.horaInicio = exigirNaoNulo(horaInicio, "Hora de início"); 
        this.grupo = exigirNaoNulo(grupo, "Grupo"); 
        this.numeroMaxParticipante = exigirNaoNulo(numeroMaxParticipante, "Número máximo de integrantes"); 
        this.peso = exigirNaoNulo(peso, "Peso");
        this.observacao = exigirTamanho(observacao, "Observação", 0 , 200); 
        this.anulada = exigirNaoNulo(anulada, "Anulada"); 
        this.motivoAnulacao =  exigirTamanho(motivoAnulacao, "Motivo Anulação", 0, 200); 
        this.idTipoFormato = exigirNaoNulo(idTipoFormato, "Formato"); 
        this.idTipoRegistroNota = exigirNaoNulo(idTipoRegistroNota, "Tipo de registro da nota"); 
        this.idTipoAbrangencia = exigirNaoNulo(idTipoAbrangencia, "Tipo de abrangência"); 
        this.idAtividade = exigirNaoNulo(idAtividade, "Atividade"); 
        this.idTipoAvaliacao = exigirNaoNulo(idTipoAvaliacao, "Tipo de avaliação"); 
        this.tempoDuracao = exigirNaoNulo(tempoDuracao, "Tempo de duração"); 
        this.idTurma = exigirNaoNulo(idTurma, "Turma"); 
        this.idDisciplina = exigirNaoNulo(idDisciplina, "Disciplina"); 
	}
}